src/App.tsx

```diff
@@ -1,32 +1,50 @@
-import { useState } from 'react';
+import { useEffect, useRef, useState } from 'react';
 
 import { FooterComponent } from './components/footer/Footer.component.tsx';
 import { HeaderComponent } from './components/header/Header.component.tsx';
 import { MainContentComponent } from './components/main-content/MainContent.component.tsx';
-import { AppContextProvider } from './context/AppContext.context.tsx';
+import { CartContextProvider } from './context/Cart.context.tsx';
+import { RouterContextProvider } from './context/Router.context.tsx';
+import { parseColorScheme, setStoredColorScheme, updateRootColorSchemeClass } from './helpers/App.helper.ts';
+import type { ColorScheme } from './types/ColorScheme.type.ts';
 
 import styles from './App.module.css';
 
 function App() {
     const FULL_NAME: string = `Roman Orlovskyi`;
     const NIK_NAME: string = `roman-orlovskyi-moc`;
 
+    const [themeMode, setThemeMode] = useState<ColorScheme>(() => parseColorScheme());
     const [isMobileMenuOpen, setIsMobileMenuOpen] = useState<boolean>(false);
 
+    const appWrapper = useRef<HTMLDivElement | null>(null);
```

Comment:
Its ok to use it here, in the educational project, in the meaning of practicing of a `useRef`.
But in the real app, would be better to make your code as simple as possible. 
And common practice in naming the variables, which contain ref, is add `ref` to them`
```suggestion
    const appWrapperRef = useRef<HTMLDivElement | null>(null);
```

Reply:
fixed